modules/mediafilterRtdProvider.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * This module adds the Media Filter real-time ad monitoring and protection module.
+ *
+ * The {@link module:modules/realTimeData} module is required
+ *
+ * For more information, visit {@link https://www.themediatrust.com The Media Trust}.
+ *
+ * @author Mirnes Cajlakovic <mcajlakovic@themediatrust.com>
+ * @module modules/mediafilterRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+import { submodule } from '../src/hook.js';
+import { logError, generateUUID } from '../src/utils.js';
+import { loadExternalScript } from '../src/adloader.js';
+import * as events from '../src/events.js';
+import CONSTANTS from '../src/constants.json';
+
+/** @const {string} MEDIAFILTER_EVENT_TYPE - The event type for Media Filter. */
+export const MEDIAFILTER_EVENT_TYPE = 'com.mediatrust.pbjs.';
+/** @const {string} MEDIAFILTER_BASE_URL - The base URL for Media Filter scripts. */
+export const MEDIAFILTER_BASE_URL = 'https://scripts.webcontentassessor.com/scripts/';
+
+export const MediaFilter = {
+  /**
+   * Registers the Media Filter as a submodule of real-time data.
+   */
+  register: function() {
+    submodule('realTimeData', {
+      'name': 'mediafilter',
+      'init': this.generateInitHandler()
+    });
+  },
+
+  /**
+   * Sets up the Media Filter by initializing event listeners and loading the external script.
+   * @param {object} configuration - The configuration object.
+   */
+  setup: function(configuration) {
+    this.setupEventListener(configuration.configurationHash);
+    this.setupScript(configuration.configurationHash);
+  },
+
+  /**
+   * Sets up an event listener for Media Filter messages.
+   * @param {string} configurationHash - The configuration hash.
+   */
+  setupEventListener: function(configurationHash) {
+    window.addEventListener('message', this.generateEventHandler(configurationHash));
+  },
+
+  /**
+   * Loads the Media Filter script based on the provided configuration hash.
+   * @param {string} configurationHash - The configuration hash.
+   */
+  setupScript: function(configurationHash) {
+    loadExternalScript(MEDIAFILTER_BASE_URL.concat(configurationHash), 'mediafilter', () => {});
```

Comment:
Did you whitelist yourself on adloader.js; if not, how did your tests pass

Reply:
I didn't notice any issues while running tests locally or observing them in pipeline, but I did go ahead and add it `src/adloader.js`. Thank you for the heads up!